src/util.ts

```diff
@@ -29,21 +29,29 @@ export function isCustomElementDecorator(node: DecoratorNode): boolean {
 /**
  * Determines if a node is an element class or not.
  *
+ * @param {Rule.RuleContext} context ESLint rule context
  * @param {ESTree.Node} node Node to test
  * @param {AST.Token=} jsdoc JSDoc to parse
  * @return {boolean}
  */
 export function isCustomElement(
+  context: Rule.RuleContext,
   node: ESTree.Node,
   jsdoc?: AST.Token | null
 ): node is ESTree.Class {
   const asDecorated = node as WithDecorators<ESTree.Node>;
+  const customElementBases: string[] = ['HTMLElement'];
+  if (context.settings.wc && context.settings.wc.elementBaseClasses) {
```

Comment:
Should this check that `context.settings.wc.elementBaseClasses` is an array?

Reply:
its a nasty `any` object so yeah fair point, people can put whatever they want in there unfortunately